### PR TITLE
feat: improve metric clarity and cleanup metric threads

### PR DESCRIPTION
**Pull request by ElastiJAM**
# PR Description

## Changes

Improves metrics clarity and adds graceful shutdown handling for background tasks to prevent loose threads.

### Metrics Improvements

- Enhanced flow lifecycle tracking (creation, active gauge, producer status: created/active/recorded/idled/dropped)
- Per-interface, per-direction packet delta tracking
- Export metrics: batch size histogram, latency tracking, improved status labels (queued/dropped/ok/error/noop)
- Channel I/O metrics: renamed `channel_sends_total` → `channel_io_total`, added exporter input tracking
- Removed redundant `SPANS_EXPORT_ERRORS` metric
- Added `MetricsSpanExporter` wrapper for automatic batch size observation

### Loose Thread Handling

- Trace ID cache cleanup task now listens to shutdown signal and exits gracefully
- Capacity shrinking task now listens to shutdown signal and exits gracefully
- Both tasks use `tokio::select!` to monitor shutdown alongside periodic intervals

Fixes #352

### Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation
- [ ] Other:

## Testing

- Metrics properly registered and exported
- Background tasks exit cleanly during shutdown
- Flow lifecycle metrics increment/decrement correctly
- Packet counting tracks deltas accurately

## Proof it works

Screenshot 2025-12-02 at 4 41 08 PM
Screenshot 2025-12-02 at 4 40 16 PM
Screenshot 2025-12-02 at 4 37 19 PM

## Checklist

- [x] I've tested my changes
- [ ] I've updated relevant documentation
- [x] My code follows the project's style (run `cargo fmt` and `cargo clippy`)
- [x] All tests pass

## Notes

- Metrics reorganized for better observability
- Background tasks properly clean up on shutdown, preventing resource leaks
- Flow active gauge handles interface migration edge cases

### CODE REVIEW

File: mermin/src/metrics/registry.rs [hunk 1]
```diff
@@ -90,10 +90,10 @@ lazy_static! {
     ).expect("failed to create channel_size metric");
 
     pub static ref USERSPACE_CHANNEL_SENDS: IntCounterVec = IntCounterVec::new(
-        Opts::new("channel_sends_total", "Total number of channel send operations")
+        Opts::new("channel_io_total", "Total number of channel I/O operations (sends)")
```
Comment (svencowart) [inline mermin/src/metrics/registry.rs]: This naming is confusing. You've changed it to `channel_io_total` — but then the ref is still called "SENDS" and you have "sends" in the description.

What is this metric reporting on? Is it the total messages sent on a channel? If so, I don't know how IO makes sense here.

File: mermin/src/span/producer.rs [hunk 2]
```diff
@@ -245,18 +246,29 @@ impl FlowSpanProducer {
...
-                        entries.scanned = scanned,
-                        entries.removed = removed,
-                        "trace ID cache cleanup completed"
-                    );
+                tokio::select! {
+                    _ = cleanup_shutdown_rx.recv() => {
+                        trace!(
+                            event.name = "trace_id_cache.cleanup_task.stopped",
+                            "trace ID cache cleanup task stopping gracefully"
+                        );
```
Comment (svencowart) [inline mermin/src/span/producer.rs]: We don't need to log this.

File: mermin/src/main.rs [hunk 3]
```diff
@@ -224,6 +226,46 @@ async fn run() -> Result<()> {
         )
         .await?;
 
+        // Log filter configuration at debug level (after tracing is initialized)
```
Comment (svencowart) [inline mermin/src/main.rs]: Instead of this, can you add a function to the Conf struct in runtime, something like, `display_conf` that prints all of it.

**Review by svencowart — CHANGES_REQUESTED** [review]
Seeing this code with our earlier slack conversation fresh in mind gives me a new theory as too why we're seeing continuous memory growth.

We have metrics with labels and those labels are frequently change based on a variety of factors and I believe that is going to cause memory usage to grow, because nothing is cleaning up stale metrics.

For example, `metrics::flow::inc_flows_created(&iface_name);` will create a new metric value for every single interface. In our netintel cluster there are several cronjobs that run every 5 minutes. For that specific metric, that means every 5 minutes we'll have at least one new metric because each cronjob will have a new unique iface name. Do that for 10ish cronjobs, that's 120 new metrics/hour and 2880 new metrics/day.  Assuming this is a u64, that's an minimum increase of 22.5 KB per day on only that one metric, and we have several of these types of metrics so I could see it ballooning rather quickly.

What is responsible for removing metrics that are no longer relevant? When does it make sense to mark a metric as irrelevant?
